### PR TITLE
Feat/508 let users change UI language

### LINKED ISSUE #508: let users change UI language
right we only support English, we would like to support more languages. To do that we need to let users change their language and we need to remember it.

### PR DESCRIPTION
**Pull request by myieye**
Resolves #508.
A first draft of #507. (I had ChatGPT translate into French and Spanish for me. Was that a terrible idea? 😆)

### CODE REVIEW

**Review by hahn-kev — CHANGES_REQUESTED** [review]
one issue I noticed is that without refreshing my jwt the language was blank on the user account page. We should probably just query the server for the user data when loading that page so that an out of date jwt won't cause any problems. It also means we can let users change data not in the jwt which makes sense.

File: frontend/src/routes/+layout.server.ts [hunk 1]
```diff
@@ -4,7 +4,8 @@ import type { LayoutServerLoadEvent } from './$types'
 import { USER_LOAD_KEY } from '$lib/user';
 import { getRootTraceparent } from '$lib/otel/otel.server'
 
-export async function load({ locals, depends, fetch }: LayoutServerLoadEvent) {
+export async function load({ locals, depends, fetch, request }: LayoutServerLoadEvent) {
+  const requestLang = request.headers.get('Accept-Language')?.split(',')[0]?.split(';')[0]?.split('-')[0]?.toLowerCase();
```
Comment (hahn-kev) [inline frontend/src/routes/+layout.server.ts:8]: added this as SSR would be wrong compared to what js determined the language should be, meaning the language would change after hydration.

Comment (myieye) [reply frontend/src/routes/+layout.server.ts:8]: Nice. I refactored it, because there's a `getLocaleFromAcceptLanguageHeader()` available

**Comment by hahn-kev** [comment]
Oh that's a good point.

Maybe just make a DTO and return that from a GQL endpoint, that should bypass the policy on the user object since it's a different object type.

**Comment by hahn-kev** [comment]
Looks good to me. @megahirt how do we feel about using machine translation for this as a first release, then we can figure out our translation pipeline and get actual humans to take a look at it.

**Comment by megahirt** [comment]
> Looks good to me. @megahirt how do we feel about using machine translation for this as a first release, then we can figure out our translation pipeline and get actual humans to take a look at it.

Yes, I am totally in favor of machine translation for UI.  We happily accept changes/PRs from enthusiastic users to fix said machine translations.

**Review by hahn-kev — APPROVED** [review]

**Comment by rmunn** [comment]
The French translation looks pretty good to me. It's possible there are some subtleties I missed from my French being rusty, but if so they can be addressed in a separate PR. The French translation is certainly good enough to merge for now, and improve later if necessary.

**Comment by myieye** [comment]
@rmunn That's encouraging to hear! It suggests that the Spanish translation also isn't garbage. 😆

**Pull request merged**